src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -146,16 +144,14 @@ public boolean canUseLocalCache() {
      */
     private boolean verifyManifestPermissions() {
         PackageManager pm = mContext.getPackageManager();
-        boolean permission = PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                "android.permission.GET_ACCOUNTS", mContext.getPackageName())
-                && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                        "android.permission.MANAGE_ACCOUNTS", mContext.getPackageName())
-                && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                        "android.permission.USE_CREDENTIALS", mContext.getPackageName());
+        boolean permission = PackageManager.PERMISSION_GRANTED == pm.checkPermission("android.permission.GET_ACCOUNTS",
```

Comment:
Manage_Accounts and Use_Credential is already deprecated in android M, so when we move to M, this will be changed. 
Well, the reason we need MANAGE_ACCOUTNS is in the case of talking to broker, adal is calling addAccounts. 
For the format change, hmm, i enable the auto-format for edited lines. I can change it back.